src/containers/PerfherderGraphContainer/index.jsx

```diff
@@ -22,19 +23,38 @@ const styles = () => ({
 class PerfherderGraphContainer extends Component {
   state = {
     data: null,
+    tooltipModel: null,
+    canvas: null,
   };
 
   async componentDidMount() {
-    this.fetchSetData(this.props);
+    this.fetchSetData(this.props).then(() => {
+      const { handleTooltip } = this;
```

Comment:
This was the only way that I could think of to call `this.handleTooltip` inside `options.tooltips.custom` without messing with the `this` that's related to the chart when that custom function is called by ChartJS

Reply:
All of this, checked. Except this line
```
const { canvas } =  self._chart;
```
The context of `self` is not inside the custom function so there's no `self._chart`.  I have another idea to avoid using `this._chart.canvas`. I'll see if I can implement it in the next pack of commits or discuss it with you first.